posthog/hogql/database/database.py

```diff
@@ -258,12 +278,18 @@ def add_warehouse_tables(self, **field_definitions: Any):
 
     def add_warehouse_self_managed_tables(self, **field_definitions: Any):
         for f_name, f_def in field_definitions.items():
-            setattr(self, f_name, f_def)
+            self.merge_or_setattr(f_name, f_def)
             self._warehouse_self_managed_table_names.append(f_name)
 
     def add_views(self, **field_definitions: Any):
         for f_name, f_def in field_definitions.items():
-            setattr(self, f_name, f_def)
+            self.merge_or_setattr(f_name, f_def)
+
+            # No need to add TableGroups to the view table names,
+            # they're already with their chained name
+            if isinstance(f_def, TableGroup):
+                continue
```

Comment:
And this is the second part of the fix - new bug after the previous one was fixed - where we don't consider `stripe` as a valid name for a view, only the full view is added here.

Reply:
do we want this for the other add_* functions?

Reply:
We could add it too, yeah, but it's not a problem because this is handled downstream. It is a fact, however, that we have some weird `stripe` (or similar) entries in the Database. As I said, it'd be ideal if we rewrote this class rather than patch it like we've been doing non-stop the last week 😢